ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/MLSdkAsyncHttpResponseHandler.java

```diff
@@ -80,6 +81,23 @@ public void onHeaders(SdkHttpResponse response) {
         SdkHttpFullResponse sdkResponse = (SdkHttpFullResponse) response;
         log.debug("received response headers: " + sdkResponse.headers());
         this.statusCode = sdkResponse.statusCode();
+        if (MapUtils.isEmpty(sdkResponse.headers())) {
```

Comment:
Should we only check the header after checking the statusCode?

Reply:
Haven't fully understand your question. But without this line, it would throw NPE when obtaining the errors in the header.

Reply:
I mean we can only check the header after checking the status code < 200 or > 300. Since most responses don't have throttling exception.
Not a blocking issue, just a suggestion

Reply:
good suggestion. Updated in the new commit.